pyrealm/constants/pmodel_const.py

```diff
@@ -117,6 +117,33 @@ class PModelConst(ConstantsClass):
     heskel_c: float = 0.0005
     """Quadratic coefficient of scaling of dark respiration (:math:`c`, 0.0005)"""
 
+    # Arrhenius values
+    arrhenius_vcmax: dict = field(
+        default_factory=lambda: dict(
+            simple=dict(ha=65330),
+            kattge_knorr=dict(
+                entropy_intercept=668.39,
+                entropy_slope=-1.07,
+                ha=71513,
+                hd=200000,
+            ),
+        )
+    )
+    """Coefficients of Arrhenius factor scaling for :math:`V_{cmax}`."""
+
+    arrhenius_jmax: dict = field(
+        default_factory=lambda: dict(
+            simple=dict(ha=43900),
+            kattge_knorr=dict(
+                entropy_intercept=659.70,
+                entropy_slope=-0.75,
+                ha=49884,
+                hd=200000,
+            ),
+        )
+    )
+    """Coefficients of Arrhenius factor scaling for :math:`J_{max}`."""
+
```

Comment:
I am sorry to be a bit stupid (Python is not my first language), but I am confused about this. Can you explain to me what the field in all those dicts and lambdas does?

Reply:
It's a peculiarity of the `dataclass` API. Default dataclass attributes get created from the class definition and so the value of mutable defaults ends up being **shared and edited across all instances of the dataclass**. So it forbids mutable defaults using the simple `attr: type = val` system and instead you have to wrap the default value in a lambda through the `field` constructor, which gives finer control over the attribute.

https://docs.python.org/3/library/dataclasses.html#mutable-default-values

Reply:
Ah. Oh dear. :laughing:

Reply:
@MarionBWeinzierl - other than that odd syntax, does this seem sensible?